ui/main.qml

```diff
@@ -151,6 +140,41 @@ StatusWindow {
                 }
             }
         }
+
+        readonly property bool macOSWindowed: Qt.platform.os === Constants.mac &&
+                                              applicationWindow.visibility !== Window.FullScreen
+
+        function restoreWindowState() {
+            switch(lastNonMinVisibility) {
+            case Window.Windowed:
+                applicationWindow.showNormal()
+                break
+            case Window.Maximized:
+                applicationWindow.showMaximized()
+                break
+            case Window.FullScreen:
+                applicationWindow.showFullScreen()
+                break
+            }
+        }
+
+        property int lastNonMinVisibility
+    }
+
+    // Save last visibility before minimize/hide so that we can bring the app in foreground in same visibility
+    onVisibilityChanged: {
+        if (applicationWindow.visibility !== Window.Minimized
+                && applicationWindow.visibility !== Window.Hidden) {
+            d.lastNonMinVisibility = applicationWindow.visibility
+        }
+    }
+
+    Binding {
+        target: Qt.application
+        property: "displayName"
+        value: d.macOSWindowed
+               ? ""
+               : qsTr("Status Desktop")
```

Comment:
Still needed? I think the displayName should stay the same

Reply:
yes I sent a screenshot in your comment above, without this in windowed mode the banners and this title overlap each other